Mathlib/Analysis/InnerProductSpace/Affine.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2025 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.Analysis.InnerProductSpace.Basic
+import Mathlib.Analysis.Normed.Group.AddTorsor
+/-
+# Normed affine spaces over an inner product space
+-/
+
+variable {𝕜 V P : Type*}
+
+section RCLike
+variable [RCLike 𝕜] [NormedAddCommGroup V] [InnerProductSpace 𝕜 V] [MetricSpace P]
+variable [NormedAddTorsor V P]
+
+open scoped InnerProductSpace
+
+theorem inner_vsub_left_eq_zero_symm {a b : P} {v : V} :
+    ⟪a -ᵥ b, v⟫_𝕜 = 0 ↔ ⟪b -ᵥ a, v⟫_𝕜 = 0 := by
+  rw [← neg_vsub_eq_vsub_rev, inner_neg_left, neg_eq_zero]
+
+theorem inner_vsub_right_eq_zero_symm {v : V} {a b : P} :
+    ⟪v, a -ᵥ b⟫_𝕜 = 0 ↔ ⟪v, b -ᵥ a⟫_𝕜 = 0 := by
+  rw [← neg_vsub_eq_vsub_rev, inner_neg_right, neg_eq_zero]
+
+theorem inner_eq_ofReal_norm_sq_left_iff {v w : V} : ⟪v, w⟫_𝕜 = ‖v‖ ^ 2 ↔ ⟪v, v - w⟫_𝕜 = 0 := by
+  rw [inner_sub_right, sub_eq_zero, inner_self_eq_norm_sq_to_K, eq_comm]
+
+theorem inner_eq_ofReal_norm_sq_right_iff {v w : V} : ⟪v, w⟫_𝕜 = ‖w‖ ^ 2 ↔ ⟪v - w, w⟫_𝕜 = 0 := by
+  rw [inner_sub_left, sub_eq_zero, inner_self_eq_norm_sq_to_K, eq_comm]
```

Comment:
These don't involve anything affine, so I expect they can go in an existing file rather than this new one.

Reply:
Good catch, moved.